frontend/src/components/collaborative/Collaborative.tsx

```diff
@@ -1,96 +1,166 @@
 import React, { memo, useState, useEffect } from 'react';
+import internal from 'stream';
+import {
+    NavigationSettings,
+    Coordinates,
+    Direction,
+    CellHintAnnotation,
+    Clue
+} from '../shared/types';
 import Crossword from '../shared/Crossword';
 import './Collaborative.css';
+import { CrosswordHint } from '../shared/CrosswordHint';
 
 export const Collaborative = memo(function Collaborative() {
+    const [error, setError] = useState(null);
+    const [isLoaded, setIsLoaded] = useState(false);
+    const [clues, setClues] = useState([]);
+    const [crosswordSize, setCrosswordSize] = useState(0);
+    const [inputCellsArray, setInputCellsArray] = useState<
+        CellHintAnnotation[][] | undefined
+    >(undefined);
 
-  const [error, setError] = useState(null);
-  const [isLoaded, setIsLoaded] = useState(false);
-  const [clues, setClues] = useState([]);
-  const [crosswordSize, setCrosswordSize] = useState(0);
-
-  // Calling sample crossword data API
-  useEffect(() => {
-    fetch("http://localhost:8080/collaborative-game/sample-crossword")
-      .then(res => res.json())
-      .then(
-        (result) => {
-          // Storing the crossword clues
-          setClues(result.clues);
-          // Storing the size of the crossword
-          setCrosswordSize(result.size);
-          // Indicates that the crossword data hass been retrieved
-          setIsLoaded(true);
-        },
-        // Handling any errors
-        (error) => {
-          setIsLoaded(true);
-          setError(error);
-        }
-      )
-  }, [])
-
-  // Array that keeps track of the cells that should be able to take in an input
-  // Initially, every cell cannot take in an input
-  let inputCellsArray:string[][]= [ 
```

Comment:
the size of this used to be hardcoded.

another thing changed is that rather than each cell being either `nonInputCell` or `inputCell`, they're either `undefined` or a `CellHintAnnotation` object.